subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationInternal.java

```diff
@@ -82,6 +82,8 @@
      */
     void beforeLocking(Action<? super ConfigurationInternal> action);
 
+    boolean isMutable();
```

Comment:
Nit: For consistency with methods like `isCanBeConsumed()`, perhaps this could be `isCanBeMutated()`.